flaxandteal/coral-arches
# Feat: hb number generation

**Description**

Implements the functionality needed to create a new HB number according to the HED format

**Test**

- Register new Wards and Districts concept
- Update collections so the new concept is loaded
- Reload the Monument model
- Reload the Revision model
- Reload webpack
- Reload plugins
- Go to the Create Building workflow, complete the first step and navigate to finish
- New Generate HB component should be at the bottom
- Check dropdown is rendering correctly
- Select an option and hit generate
- The new HB Number should appear in the disabled input
- Click save at the bottom of the workflow
- Check the resource model HA References the HB number should be present
- Check the resource model Wards and Districts the HB Number should be present
- Because this was the first reference number to be submitted it should be in Display Name and toggled on
- Start the designation workflow with the same resource you created the Hb Number on
- A Revision copy should be created with all the same resource models checks from a couple steps ago
- Confirm completing the Designation workflow will merge the same data back into the original without any issues
